docs/markdown/Reference-tables.md

```diff
@@ -15,7 +15,8 @@ These are return values of the `get_id` (Compiler family) and
 | clang-cl  | The Clang compiler (MSVC compatible driver) | msvc |
 | dmd       | D lang reference compiler        |                 |
 | emscripten| Emscripten WASM compiler         |                 |
-| flang     | Flang Fortran compiler           |                 |
+| flang     | Flang Fortran compiler (LLVM-based) |              |
+| flang     | Classic Flang Fortran compiler   |                 |
```

Comment:
The values here are the same, they probably should not be.

Reply:
And also "flang-new" is not really a great name. Maybe something like "flang-llvm"?

Reply:
> The values here are the same, they probably should not be.

Unfortunately, they are (at least from the POV of the name of the binary). Of course, from the POV of meson they can and should be distinguished.

> And also "flang-new" is not really a great name. Maybe something like "flang-llvm"?

That's the "official" upstream name of the binary. They've been promising a rename to "flang" for a long time, and it's bound to happen "soon".

Reply:
It shouldn't matter what the official name of the binary is. E.g. the new Intel compilers are referred to by Meson as `intel-llvm`, not `icx`/`ifx` etc.. It looks to me like `flang-llvm` is nicely consistent with `intel-llvm`.

Reply:
Yeah. The issue is that this table needs to describe how people figures out which compiler they are using, via compiler.get_id(), so they can write meson.build files that handle conditional logic depending on whether the build is using flang from llvm or flang classic.

Reply:
> This table has nothing to do with executable name lookups.

Yeah, I get that, I just mentioned it since @jpakkane brought up `flang-new`. Happy to rename this `flang-llvm` for the compiler-id.

Reply:
If that is what upstream calls it, then let's go with it.

Reply:
Not sure what you're referring to with "that". Upstream calls itself "flang", but has a current temporary (see this [thread](https://discourse.llvm.org/t/proposal-rename-flang-new-to-flang/69462/52), for example) name of "flang-new" for the binary. That rename is pretty much set in stone, but the exact timing is unclear (IMO the fact that it hasn't happened yet is a subversion of the spirit of the conflict resolution they went through, as all publicly declared milestones were met).

That aside, the output of `<binary> --help` starts with `OVERVIEW: flang LLVM compiler`. The original flang (see [here](https://github.com/flang-compiler/flang)) is now called "classic flang" by the way, although "original" is stretching things, in the sense that even classic flang was based on PGI's pgfortran (which looks to have a `pgi` compiler-id in meson).

Not sure if there's a way to deprecate bare "flang" as a compiler-id, but perhaps the cleanest setup would be:
* `classic-flang` (== current `flang`; alternatively as an alias?)
* `llvm-flang` (this PR)
* `armltdflang` (pre-existing)

Reply:
I don't really think we can, because the compiler ID is intended to be used like this:

```meson
fc = meson.get_compiler('fortran')

if fc.get_id() == 'flang'
    message('running code that is specific to classic flang')
elif fc.get_id() == 'pgi'
    message('running code that is specific to PGI pgfortran')
elif fc.get_id() == 'llvm-flang'
    message('running code that is specific to LLVM 19 integrated flang')
endif
```

We can't return both values. CMake's "policies" would allow picking and choosing what CMake returns in such a situation, but meson made the decision to not implement anything like a CMake policy system, because we felt that it tends to lead to spaghetti code that is hard to understand or tell how it behaves.

Reply:
OK, so then we keep `flang` as is for classic flang, and people just need to decide on a name for the llvm variant. Feel free to tell me directly, or respond according to the emojis below:
* 👍 `flang-llvm`
* 🚀 `llvm-flang`
* 👀 other

Reply:
I've gone with `llvm-flang` now, which I already had in use in various places; happy to change it though.